### DISCUSSION TITLE
AOT ClassConstructorRunner

### DISCUSSION THREAD

**Discussion by a1443304527** [AvaloniaUI/Avalonia#17109] [Q&A]
### Describe the bug

 Application: BasicMvvmSample.exe CoreCLR Version: 9.0.0-rc.1.24431.7 Description: The process was terminated due to an unhandled exception. Exception Info: System.TypeInitializationException: A type initializer threw an exception. To determine which type, inspect the InnerException's StackTrace property. at System.Runtime.CompilerServices.ClassConstructorRunner.EnsureClassConstructorRun(StaticClassConstructionContext*) + 0xf3 at System.Runtime.CompilerServices.ClassConstructorRunner.CheckStaticClassConstructionReturnNonGCStaticBase(StaticClassConstructionContext*, IntPtr) + 0x8 at Avalonia.Skia.SkiaPlatform.Initialize(SkiaOptions) + 0x11c at Avalonia.SkiaApplicationExtensions.<>c.<UseSkia>b__0_0() + 0x4d at Avalonia.AppBuilder.SetupUnsafe() + 0x2f at Avalonia.AppBuilder.Setup() + 0x39 at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder, String[], Action`1) + 0x27 at BasicMvvmSample.Program.Main(String[] args) + 0x16  

### To Reproduce

![image](https://github.com/user-attachments/assets/4ae4c01a-ab8a-492b-8c0c-bf51d7981aa7)

### Avalonia version

11.1.3

### OS

Windows

**Comment by timunie** [comment]
@a1443304527 that issue description is not really helpful. If you need AOT compilation make sure your project is trimming friendly.

**Comment by kyuranger** [comment]
It seems you are using a preview version .Net. Try replacing with a LTS .Net version (like .net 8)
